Docs for Heartcore Persisted queries (#6554)
umbraco/UmbracoDocs: umbraco-heartcore/api-documentation/graphql/persisted-queries.md

---
description: Documentation for Persisted queries in Umbraco Heartcore
---

# Persisted queries

Persisted queries allow you to store GraphQL queries on the server. This enables clients to execute them by referencing an alias rather than sending the full query each time. this approach streamlines client-server communication and enhances security.

## Why use persisted queries?

* Reducing the payload size

>     By sending only the alias and necessary variables, you minimize the data transmitted over the network. This reduction in payload size leads to faster request times and is particularly beneficial for mobile applications or environments with limited bandwidth.

Reviewer: ```suggestion
    * You minimize the data transmitted over the network by sending only the alias and necessary variables. This reduction in payload size leads to faster request times and is particularly beneficial for mobile applications or environments with limited bandwidth.
```

* Enhanced Security

    Persisted queries provide improved security by ensuring that only predefined, server-stored GraphQL queries can be executed, preventing clients from running arbitrary or malicious queries.

>     However, to fully benefit from this security enhancement, it is essential that the content delivery API is set to private, and GraphQL is configured to only allow persisted queries. Without these settings in place, the security advantages of using persisted queries are not realized.

Reviewer: ```suggestion
To fully benefit from this, the content delivery API must be set to private, and GraphQL must be configured only to allow persisted queries. Without these settings in place, the security advantages of using persisted queries are not realized.
```

---
description: Documentation for Persisted queries in Umbraco Heartcore
---

# Persisted queries

> Persisted queries allows you to store GraphQL queries on the server, enabling clients to execute them by referencing an alias rather than sending
> the full query each time. this approach streamlines client-server communication and enhances security.

Reviewer: ```suggestion
Persisted queries allow you to store GraphQL queries on the server. This enables clients to execute them by referencing an alias rather than sending the full query each time. this approach streamlines client-server communication and enhances security.
```
